utils/fs/directory.go

```diff
@@ -46,6 +48,18 @@ func LastModified(search *LastModifiedSearch) (time.Time, error) {
 			}
 			return err
 		}
+		// Append the cwd if the paths are relative. This is needed because
+		// source/artifact handling (`filepath.Rel()`) needs absolute paths to work properly.
+		cwd, err := execenv.ValueFromFilesystem("cwd", "")
+		if err != nil {
+			return err
+		}
+		if !strings.HasPrefix(rootPath, "/") {
+			rootPath = cwd + "/" + rootPath
+		}
+		if !strings.HasPrefix(filePath, "/") {
+			filePath = cwd + "/" + filePath
+		}
```

Comment:
Couple thoughts here:

1. I guess this will break on windows, this should probably use https://golang.org/pkg/path/filepath/#IsAbs
2. Since `rootPath` never changes I think we can do this work before the `walker` function instead of inside `walker`.

I believe if `rootPath` is absolute that `filePath` will always be absolute, but a test for each of those cases should confirm.

Reply:
1. yes, i didn't think about that. Using `filepath.IsAbs()` and `os.PathSeperator` 
2. you are right, will move that one outside. 

Btw: the reason why the tests always succeeded with those changes was that `tmpdir := os.NewDir` and `tmpdir.Path()` always returns absolute paths. I will add a directory test for relative paths.